draft-ietf-tls-extended-key-update.md

```diff
@@ -534,37 +440,54 @@ Auth | {CertificateVerify}
   (with key_share))]
                                <-------- [ExtendedKeyUpdate(response
                                            (accepted, with key_share))]
- [ExtendedKeyUpdate(new_key_update)]     -------->
-                                         <--------
+										# Server derives new secrets
+# Client derives new secrets			
+ [ExtendedKeyUpdate(new_key_update)]
+                               -------->
+# Client updates SEND keys here
+                                    # Server updates RECEIVE keys here
+   							   <--------
                                     [ExtendedKeyUpdate(new_key_update)]
+									# Server updates SEND keys here
+
+# Client updates RECEIVE keys here
 ~~~
-{: #fig-key-update2 title="Extended Key Update Example Exchange."}
+{: #fig-key-update2 title="Extended Key Update Example."}
 
 #  DTLS 1.3 Considerations
 
 Unlike TLS 1.3, DTLS 1.3 implementations must take into account that handshake
 messages are not transmitted over a reliable transport protocol.
 
+Due to the possibility of an `ExtendedKeyUpdate(new_key_update)` message being
+lost and thereby preventing the sender of that message from updating its keying
+material, receivers MUST retain the pre-update keying material until receipt
+and successful decryption of a message using the new keys.
+
+Due to loss and/or reordering, DTLS 1.3 peers MAY receive a record with an
+older epoch than the current one. They SHOULD attempt to process such records
+for that epoch but MAY opt to discard such out-of-epoch records.
+
 The exchange has the following steps:
 
-1. The initiator sends an `ExtendedKeyUpdate(request)` message, which contains a key
-   share. While an extended key update is in progress, the initiator MUST NOT
+1. The initiator sends an `ExtendedKeyUpdate(request)` message, which contains a
+   key share. While an extended key update is in progress, the initiator MUST NOT
    initiate further key updates.
 
 2. On receipt of the `ExtendedKeyUpdate(request)`, the responder either accepts
    or declines the request. If the responder accepts the request, it sets the
-   status in `ExtendedKeyUpdate(response)` to `accepted` and includes its own key
-   share. While an extended key update is in progress, the responder MUST NOT
-   initiate further key updates. If the responder declines the request, it sets
-   the status accordingly and does not include a key share. Declining the request
-   aborts the exchange.
+   status in `ExtendedKeyUpdate(response)` to `accepted`, includes its own key
+   share, and sets the local variable `accepted=1`. While an extended key update
+   is in progress, the responder MUST NOT initiate further key updates. If the
+   responder declines the request, it sets the status accordingly and does not
+   include a key share. Declining the request aborts the exchange.
 
 3. If the status in `ExtendedKeyUpdate(response)` was set to `accepted`,
```

Comment:
Step 3 does not specify what happens when the status in ExtendedKeyUpdate(response) is not set to "accepted". 
I don’t see a need for Step 3, since the responder must send an ExtendedKeyUpdate(response) regardless of whether the status is "accepted" or not.